src/jit/lir.cpp

```diff
@@ -552,10 +492,39 @@ GenTree* LIR::Range::FirstNonPhiOrCatchArgNode() const
             continue;
         }
 
-        break;
+        return node;
+    }
+
+    return nullptr;
+}
+
+//------------------------------------------------------------------------
+// LIR::Range::PhiNodes: Returns the range of phi nodes inside this range.
+//
+LIR::ReadOnlyRange LIR::Range::PhiNodes() const
+{
+    GenTree* firstNonPhiNode = FirstNonPhiNode();
+    if (firstNonPhiNode == m_firstNode)
+    {
+        return ReadOnlyRange();
     }
 
-    return node;
+    return ReadOnlyRange(m_firstNode, firstNonPhiNode == nullptr ? m_lastNode : firstNonPhiNode->gtPrev);
+}
```

Comment:
With LastPhiNode(), this could be nicely symmetric to NonPhiNodes():

    GenTree* lastPhiNode = LastPhiNode();
    if (lastPhiNode == nullptr)
    {
        return ReadOnlyRange();
    }

    return ReadOnlyRange(m_firstNode, lastPhiNode);